proposals/07-datastore-optimization.md

```diff
@@ -0,0 +1,101 @@
+# Hi-speed Chain Store
+
+Authors: @hannahhoward @acruikshank @willscott @gammazero @mvdan
+
+Initial PR: https://github.com/protocol/web3-dev-team/pull/7
+
+
+## Purpose &amp; impact 
+#### Background &amp; intent
+
+Chain syncing in Filecoin is painfully slow. This affects the whole product -- system requirements to run even a client node are very high, and gas prices and message limits are limited by the speed we can sync the chain. One huge source of this time suck is simply reading data in and out of our datastore.
+
+We have approached data persistence through using off the shelf key-value stores in our stack. Long term, many aspects of our stack are limited by the lack of a datastore/database designed specifically for our data format, IPLD (see this [overview](https://docs.google.com/document/d/1GxUNUaBfaY4HneWCOfyPcyWG_14t-vnsy3NbQRF4Fdc/edit) and [analysis](https://docs.google.com/document/d/16_8TEslIY__54x2lvMNu5dLL0BSAJ2GxkZ9w3p8sUys/edit)). But in the near term, we have several extremely promising alternative key/value store solutions that can deliver immediate concrete improvements to chain performance.
+
+The aim here is to finalize existing explorations to deliver at least a 50% improvement to chain performance.
+
+#### Assumptions &amp; hypotheses
+
+* Datastore performance affects chain sync speed (we have experimental evidence this is true, but not indications of how much)
+* Prototyped solutions that demonstrate improvement as prototypes will not get slowed down a lot in getting to production readiness
+* Chain performance raises the barrier to entry and creates negative characteristics that cause people to leave Filecoin (see:  high gas fees, long time to get up and running without snapshot)
+* A 50% improvement that isn’t an order of magnitude improvement is still worth pursuing, especially if we have a clear path to success
+* Existing work that hasn’t been finished and benchmarked will deliver at least modest improvements
+
+
+#### User workflow example
+_How would a developer or user use this new capability?_
+<!--(short paragraph)-->
+
+#### Impact
+
+🔥🔥
+
+Chain speed breaks everything. Even 50% improvement has a big impact on everyone using filecoin.
+
+#### Leverage
+
+🎯🎯
+
+We are specifically picking this as an easy win on the datastore improvement track. However, it might be something we bring to IPFS eventually, and it will set us up to build more specialized IPLD solutions.
+
+#### Confidence
+
+3 - All user reports that the hardware requirements for running a full lotus node are a major pain point -- if we could drop those requirements, even a little, it would make a big difference
```

Comment:
Totally agree with this confidence in the problem - but as per the really good flags above, we aren’t quite as confident in our proposed solution